### ISSUE TITLE
aws-ec2 : Lookup funciton for transit gateway

### ISSUE THREAD

**Issue by hrider** [aws/aws-cdk#32091]
### Describe the feature

Provide a lookup method for transit gateway. When Transit gateway attachments are provisioned on multi accounts, tags do not carry over. By having a lookup method, we can add the tags programatically.

### Use Case

When Transit gateway attachments are provisioned in multi account setup, tags do not carry over into the owner account. By having a lookup method, we can add the tags programmatically.

### Proposed Solution

Provide lookup method for transit gateway.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.160.0

### Environment details (OS name and version, etc.)

mac sonoma 14.7

**Comment by pahud** [comment]
Can you share more details and if possible, give me some sample code snippets of your preferred design?

This sound like using a fromLookup() method to query existing transit gateways using SDK calls and add tags to them?

**Comment by hrider** [comment]
You are correct. I am seeking sth similar to IVPC. ec2.Vpc.fromLookup(this, 'vpc', { vpcId: 'vpc-xxx' });
